app/livechat/server/lib/Livechat.js

```diff
@@ -461,25 +463,27 @@ export const Livechat = {
 		return rcSettings;
 	},
 
-	saveRoomInfo(roomData, guestData) {
+	saveRoomInfo(roomData, guestData, userId) {
 		const { livechatData = {} } = roomData;
 		const customFields = {};
 
-		const fields = LivechatCustomField.find({ scope: 'room' });
-		fields.forEach((field) => {
-			if (!livechatData.hasOwnProperty(field._id)) {
-				return;
-			}
-			const value = s.trim(livechatData[field._id]);
-			if (value !== '' && field.regexp !== undefined && field.regexp !== '') {
-				const regexp = new RegExp(field.regexp);
-				if (!regexp.test(value)) {
-					throw new Meteor.Error(TAPi18n.__('error-invalid-custom-field-value', { field: field.label }));
+		if (userId && hasPermission(userId, 'edit-livechat-room-customfields')) {
```

Comment:
We can't apply this check this way. This is method is called from non-authenticated users, like the Livechat visitors through the Livechat widget API.
So, you could do something like this:

```suggestion
		if (!userId || hasPermission(userId, 'edit-livechat-room-customfields')) {
```

What do you think?
Also, you need to do the same here:
https://github.com/RocketChat/Rocket.Chat/blob/eef240e624d88a6dfffdf03aa8588f1896806716/app/livechat/server/lib/Livechat.js#L294